### PR TITLE
First fix for #1498 (disabling newrelic psycopg2 import hook approach)

### LINKED ISSUE #1498: Dev & Staging v2.5.0 - Exceptions thrown when calling API methods
Looks like a problem with `djorm_pgfulltext`. Logs below:

```txt
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/home/vcap/app/api/views.py", line 173, in get
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     contracts_all = self.get_queryset(request.query_params, wage_field)
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/home/vcap/app/api/views.py", line 205, in get_queryset
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     return get_contracts_queryset(request, wage_field)
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/home/vcap/app/api/views.py", line 100, in get_contracts_queryset
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     contracts = contracts.multi_phrase_search(qs)
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/home/vcap/app/contracts/models.py", line 136, in multi_phrase_search
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     return self.search(convert_to_tsquery_union(queries), raw=True)
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/app/.heroku/python/lib/python3.6/site-packages/djorm_pgfulltext/models.py", line 315, in search
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     "%s('%s', %s)" % (function, config, adapt(query))
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR   File "/app/.heroku/python/lib/python3.6/site-packages/djorm_pgfulltext/utils.py", line 10, in adapt
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR     a.prepare(connection.connection)
2017-03-16T22:54:03.13-0500 [APP/PROC/WEB/0]ERR TypeError: argument 1 must be psycopg2.extensions.connection, not ConnectionWrapper

2017-03-16T22:54:03.22-0500 [RTR/1]      OUT calc-staging.app.cloud.gov - [2017-03-17T03:54:03.123+0000] "GET /api/rates/?histogram=12&q=te&min_experience=0&max_experience=45&sort=current_price&query_type=match_all&experience_range=0%2C45 HTTP/1.1" 500 0 20535 "https://calc-staging.app.cloud.gov/?q=te" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.98 Safari/537.36" "127.0.0.1:8576" "10.10.3.7:63976" x_forwarded_for:"72.48.211.158, 72.48.211.158, 54.208.160.151" x_forwarded_proto:"https" vcap_request_id:"bcfed5b7-5af3-489f-4be5-134d6d04ce7e" response_time:0.100149665 app_id:"db7756dc-7ca3-417a-aeb2-bb7e0b1e9286" app_index:"0"
```

### PR DESCRIPTION
**Pull request by toolness**
This is a workaround to fix #1498, based on a [comment on a gist from someone who appears to be a NewRelic employee](https://gist.github.com/mjallday/3d4c92e7e6805af1e024#gistcomment-1379981):

> FWIW, the issue is caused by psycopg2 bypassing its own public psycopg2.extensions module to access register_type(), going direct to the C extension module. A workaround should be able to be setup from the agent configuration file to remap what our instrumentation is applied to, until we can add a more permanent fix into the agent package itself. 

Blerg. I guess the downside is that we lose database instrumentation but I guess I've never actually used it (or NewRelic) anyways so whatever.

### CODE REVIEW

File: hourglass/wsgi.py [hunk 1]
```diff
@@ -24,11 +27,12 @@
 
 nr_license = os.environ.get('NEW_RELIC_LICENSE_KEY')
 nr_app_name = os.environ.get('NEW_RELIC_APP_NAME')
+nr_ini_path = os.path.join(BASE_DIR, 'newrelic.ini')
 if nr_license and nr_app_name:
     nr_settings = newrelic.agent.global_settings()
     nr_settings.license_key = nr_license
     nr_settings.app_name = nr_app_name
-    newrelic.agent.initialize()
+    newrelic.agent.initialize(nr_ini_path)
```
Comment (toolness) [inline hourglass/wsgi.py]: Um, so this is kind of weird--apparently we were never actually loading our own `newrelic.ini` file?  I hope the rest of our `newrelic.ini` is actually valid, then... Bleh.

**Comment by toolness** [comment]
To test this out, add the following to your `.env` file:

```
NEW_RELIC_LICENSE_KEY=bleh
NEW_RELIC_APP_NAME=whatevs
```

Then comment out the new lines added to `newrelic.ini` in this PR.

Then start up your server. You should be able to reproduce #1498 by typing into the search field on CALC's front page.

Then uncomment the new lines in `newrelic.ini` and reboot the server (either by doing it manually or force-saving a file like `settings.py`).  The bug should be fixed!

File: hourglass/wsgi.py (hunk 1 above)
Comment (jseppi) [inline hourglass/wsgi.py]: We actually specify the config file in our manifests via the `NEW_RELIC_CONFIG_FILE` environment variable. See https://github.com/18F/calc/blob/develop/manifests/manifest-dev.yml#L19 for example.

**Comment by jseppi** [comment]
I'm a little wary of turning off instrumentation. We luckily haven't had to use it much because things are generally working well - it's when things go bad that the instrumentation will come in handy. 

Do you think we'll be able to re-enable it after moving to the Django 1.10/11 native text search (ref https://github.com/18F/calc/issues/1346)?

**Comment by jseppi** [comment]
Also you should poke around in NR. It's kinda fun!

**Comment by jseppi** [comment]
We could also contact NR support. They're pretty responsive.

File: hourglass/wsgi.py (hunk 1 above)
Comment (toolness) [inline hourglass/wsgi.py]: O ok. Should we actually just pass it into `initialize()` though, to ensure better dev/prod parity?

**Comment by toolness** [comment]
O good point about the ol' disabling stuff. I've filed https://github.com/18F/calc/pull/1500 as an alternative fix.

Want to give us a tour of NR at the next coworking? 😁

**Comment by toolness** [comment]
Ended up going with #1502 instead!

**Pull request closed without merging**